modules/database/database.bal

```diff
@@ -1353,3 +1338,49 @@ public isolated transactional function saveTimelineSubstepInputData(int stepId,
         _ = check experimentDB->execute(`INSERT INTO SubstepData (stepId, substepNr, dataId, relationType) VALUES (${stepId}, ${substepNr}, ${experimentData.dataId}, ${"input"});`);
     }
 }
+
+# # Filter functions
+
+# Build filter query fragment to find matches in name, type and content of experiment data.
+#
+# + experimentId - Experiment id
+# + search - Search string to match
+# + return - Return filter query fragment
+public isolated function experimentDataFilter(int experimentId, string search) returns sql:ParameterizedQuery {
+    sql:ParameterizedQuery filter = `  WHERE experimentId = ${experimentId} `;
+    if search != "" {
+        string searchString = "%" + search + "%";
+        return sql:queryConcat(filter, ` AND ((name LIKE ${searchString}) OR (type  LIKE ${searchString}) OR (contentType LIKE ${searchString})) `);
+    }
+    return filter;
+}
+
+# Build filter query fragment or timeline step list queries
+#
+# + experimentId - Experiment id
+# + plugin\-name - Plugin name filter  
```

Comment:
you do not need to keep using - in identifiers for functions that are not mapped to REST resource URLs. Just use camel case...